### ISSUE TITLE
Screenreader speaks unnecessary information when opening DatePicker

### ISSUE THREAD

**Issue by CobaltBlack** [microsoft/fluentui#20075]
### Environment Information

- **Package version(s)**: 8.36.0, 7.177.1
- **Browser and OS versions**: Edge, Windows 10

### Describe the issue:

When DatePickeropened, JAWS reads out all weekday headers and abbreviated month names. See attached screenshot below of the speech history. Note the "S M T W T F S Jan Feb...". 

Reading out the abbreviated weekday and months is not useful to the screen reader user.

![image](https://user-images.githubusercontent.com/6284654/135696220-3df6ee32-124e-46b3-8c18-47f7c3e725cf.png)

### Please provide a reproduction of the issue in a codepen:

The issue is reproducible in the demo site, using the first datepicker:
https://fabricweb.z5.web.core.windows.net/oufr/8.36.0/#/examples/datepicker
https://fabricweb.z5.web.core.windows.net/oufr/7.177.1/#/examples/datepicker

#### Actual behavior:

Screen reader will read "S M T W T F S Jan Feb..." when opening the DatePicker.

#### Expected behavior:

Screen reader should not read "S M T W T F S Jan Feb..." when opening the DatePicker.
